src/editor/components/scenegraph/Toolbar.js

```diff
@@ -137,8 +102,13 @@ export default class Toolbar extends Component {
     }
   };
 
-  cloudSaveAsHandler = async () => {
-    this.cloudSaveHandler({ doSaveAs: true });
+  cloudSaveHandlerWithImageUpload = async (doSaveAs) => {
```

Comment:
changed cloudSaveAs to be a wrapper of cloudsave w/ uploading an image, since image uploads happen primarily on save as (unless something goes wrong)